include/swift/AST/Evaluator.h

```diff
@@ -288,8 +286,6 @@ class Evaluator {
            typename std::enable_if<!Request::isEverCached>::type * = nullptr>
   llvm::Expected<typename Request::OutputType>
   operator()(const Request &request) {
```

Comment:
This invariant needs a static assert.

Reply:
Latest version no longer has this invariant. Dependency sources that are uncached are now supported again in service of ASTLoweringRequest.

Reply:
I can't quite tell from the diff, but are we still recording and replaying here without this?